rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+        else:
+            querystr = format_query_indentation(QUERY_STAKE)
+        return self.graph.query(
+            querystr=querystr,
+            param_types=param_types,
+            param_values=param_values,
+        )
+
+    def get_history(
+        self,
+        addresses: List[ChecksumEthAddress],
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+    ) -> Dict[ChecksumEthAddress, Dict[str, List[LiquityEvent]]]:
+        query = self._get_raw_history(addresses, 'trove')
+        staked = self._get_raw_history(addresses, 'stake')
+        result: Dict[ChecksumEthAddress, Dict[str, List[LiquityEvent]]] = defaultdict(lambda: defaultdict(list))  # noqa: E501
+        for trove in query['troves']:
+            owner = to_checksum_address(trove['owner']['id'])
+            for change in trove['changes']:
+                try:
+                    timestamp = change['transaction']['timestamp']
+                    if timestamp < from_timestamp:
+                        continue
+                    if timestamp > to_timestamp:
+                        break
+                    operation = TroveOperation.deserialize(change['troveOperation'])
+                    collateral_change = deserialize_optional_fval(change['collateralChange'], 'collateralChange', 'liquity')  # noqa: E501
+                    debt_change = deserialize_optional_fval(change['debtChange'], 'debtChange', 'liquity')  # noqa: E501
+
+                    event = LiquityTroveEvent(
+                        kind='trove',
+                        tx=change['transaction']['id'],
+                        address=owner,
+                        timestamp=timestamp,
+                        debt_after=deserialize_optional_fval(change['debtAfter'], 'debtAfter', 'liquity'),  # noqa: E501
+                        collateral_after=deserialize_optional_fval(change['collateralAfter'], 'collateralAfter', 'liquity'),  # noqa: E501
+                        debt_delta=debt_change,
+                        collateral_delta=collateral_change,
+                        trove_operation=operation,
+                    )
+                    result[owner]['trove'].append(event)
+                except KeyError as e:
+                    self.msg_aggregator.add_warning(
+                        f'Ignoring Liquity Trove event in Liquity. '
+                        f'Failed to decode remote information. {str(e)}.',
+                    )
+                    continue
+
+        for stake in staked['lqtyStakes']:
+            owner = to_checksum_address(stake['id'])
+            for change in stake['changes']:
+                try:
+                    timestamp = change['transaction']['timestamp']
+                    if timestamp < from_timestamp:
+                        continue
+                    if timestamp > to_timestamp:
+                        break
+                    operation_stake = LiquityStakeEventType.deserialize(change['stakeOperation'])
```

Comment:
same comment in this try/catch for catching `DeserializationError` from all the `deserialize_` functions